### PR TITLE
Use identity-hostdata, remove dotenv

### PR DESCRIPTION
**Pull request by zachmargolis**
The only configs depend on the which deploy environment,
and not on other configs, so we can use a single saml_settings
and template it.

### CODE REVIEW

**Comment by monfresh** [comment]
I don't think this is related to this PR since I'm getting the same error on the `master` branch, but when I sign in at the IdP, I get this error after being redirected to the SP:
```
OneLogin::RubySaml::ValidationError at /consume
An EncryptedAssertion found and no SP private key found on the settings to decrypt it. 
Be sure you provided the :settings parameter at the initialize method
```

File: app.rb [hunk 1]
```diff
@@ -75,12 +75,12 @@ def auth_server_uri
   end
 
   post '/consume/?' do
-    response = OneLogin::RubySaml::Response.new(params[:SAMLResponse])
+    response = OneLogin::RubySaml::Response.new(
+      params.fetch(:SAMLResponse), settings: saml_settings
```
Comment (monfresh) [inline app.rb]: This should be `'SAMLResponse'`, as a String, not a Symbol. I just tested it locally.

Comment (monfresh) [reply app.rb]: The only thing that I can think of that would explain how this ever worked is that the assertion was not encrypted at some point.  The only scenario where the `ruby-saml` gem expects the settings to be available during initialization is if the assertion is encrypted. If the assertion is not encrypted, you can set the settings after the fact and then call `is_valid?`.

Comment (brodygov) [reply app.rb]: Thanks, good catch.

**Comment by brodygov** [comment]
Tested this in the Brody environment, fixed the bug Moncef identified, and things appear to be working normally now.

**Pull request merged**